src/core/dynamic_system.jl

```diff
@@ -1,60 +1,95 @@
 """
-function sesolve(
-        H::AbstractOperator, 
-        ψ_0::Ket{S}, 
-        t_range::StepRangeLen; 
-        e_ops::Vector{T} where {T<:AbstractOperator}=(T)[], 
-    )::Tuple{Vector{Ket{S}},Matrix{<:Real}} where {S<:Complex}
-    Hamiltonian(t)=H
-    sesolve(Hamiltonian, ψ_0, t_range; e_ops=e_ops)
+Base.@kwdef struct LindbladProblem{T<:DenseOperator}
+    H::Function
+    init_state::T
+    tspan::Tuple{Float64,Float64}
+    e_ops::Vector{T}
+    c_ops::Vector{T}
 end
 
+"""
+    sesolve(problem::ShrodingerProblem; kwargs...)
+Solves the Shrodinger equation:
+
+``\\frac{d \\Psi}{d t}=-i \\hat{H}\\Psi``
+
+and returns a tuple correponding the time instance vector, the corresponding wavefunction Ket, and a Vector of observables evaluated at each time step. 
+
+**Fields**
+- `problem` -- An object of type ShrodingerProblem that defines the problem to be solved. 
+- `is_hamiltonian_static` -- A Bool variable indicating whether the Hamiltonian operator changes with time or not. Default value is false. If true, the solver can have significant performance boost.
+- `kwargs` -- list of keyword arguments to be passed to the ODE solver. See (https://docs.sciml.ai/DiffEqDocs/stable/basics/common_solver_opts/#solver_options).
+"""
 function sesolve(
-        H::Function, 
-        ψ_0::Ket{S}, 
-        t_range::StepRangeLen; 
-        e_ops::Vector{T} where {T<:AbstractOperator}=(T)[], 
-    )::Tuple{Vector{Ket{S}},Matrix{<:Real}} where {S<:Complex}
-    dpsi_dt(t,ψ) = -im*H(t)*ψ
-    y=rungekutta2(dpsi_dt, ψ_0, t_range)
-    n_t = length(t_range)
-    n_o = length(e_ops)
-    observable=zeros(n_t, n_o) 
-    for iob in 1:n_o
-        for i_t in 1:n_t
-            observable[i_t, iob] = real(expected_value(e_ops[iob], y[i_t]))
+    problem::ShrodingerProblem;
+    is_hamiltonian_static::Bool = false,
+    kwargs...
+)
+        
+        dψ_dt_0 = -im*sparse(problem.H(0)).data
+        function Hamiltonian!(dψ_v,ψ_v,p,t)
+            if (is_hamiltonian_static)
+                mul!(dψ_v,dψ_dt_0,ψ_v)
+            else
+                mul!(dψ_v,sparse(-im*problem.H(t)).data,ψ_v)
+            end
+            nothing
         end
-    end        
-    return (y, observable)
+        prob = OrdinaryDiffEq.ODEProblem(Hamiltonian!,problem.init_state.data,problem.tspan)
+        sol=OrdinaryDiffEq.solve(prob,OrdinaryDiffEq.AutoTsit5(OrdinaryDiffEq.Rosenbrock23(autodiff=false));kwargs...) #The default algo is Tsit5 but we switch to Rosenbrock23 for stiff problems
+        y=Vector{typeof(problem.init_state)}()
+        for v in sol.u
+            push!(y,Ket(v))
+        end
+        
+        n_t = length(sol.t)
+        n_o = length(problem.e_ops)
+        observable=zeros(n_t, n_o) 
+        for iob in 1:n_o
+            for i_t in 1:n_t
+                observable[i_t, iob] = real(expected_value(problem.e_ops[iob], y[i_t]))
+            end
+        end        
+        return (t=sol.t,u=y,e=observable)
```

Comment:
See comment regarding storage of output in `mesolve` comment

Reply:
I'm not sure if I found that comment.

Reply:
It seems it went missing somewhere, but I'll repeat it here.

The benefit of passing in observables is that we can limit what we store in memory. Storing the entire system's state (statevector in the Schrodinger case and density matrix in the Lindblad case) for all time points is going to be very memory intensive as we try to scale up the simulation size.

I think we should either use the `Integrator` interface  (https://docs.sciml.ai/DiffEqDocs/stable/basics/integrator/) and only store and return the observables. Alternatively, we can only return the observables and add an issue that we should improve the function to use less memory.

Reply:
Fair point. I suggest we add an issue and resolve this in another iteration.,

Reply:
I've added an issue: https://github.com/anyonlabs/Snowflake.jl/issues/240